### ISSUE TITLE
TiddlyMap destroys tiddler body content when indexing

### ISSUE THREAD

**Issue by sycom** [felixhayashi/TW5-TiddlyMap#243]
If you have manually created  `.tid` tiddlers and launch a node tiddlywiki server, tiddlymap will index all tiddlers that don't have the `tmap.id` field. But I recently experienced that it also deleted all of body content (only body: all fields are preserved. Except `modified`)

**Comment by felixhayashi** [comment]
Strange. Thanks for reporting @sycom. I'll have a look.

**Comment by felixhayashi** [comment]
Hi @sycom, I wasn't able to reproduce this bug. maybe it got fixed with v0.12.0 as I changed much of the previous codebase…

**Comment by sycom** [comment]
Hi @felixhayashi , thanks for looking out.

Also tried to reproduce the bug from scratch. Was not so easy and for now I'm not sure (see at the end). Might be a misuse of TiddlyWiki server or a problem with save process and not with your plugin.

So step by step : 

1.  Created a fresh Test folder for my wiki, with a very simple tiddlywiki.info 
````
{
   "description": "Fresh wiki from scratch",
   "plugins": [
      "tiddlywiki/tiddlyweb",
      "tiddlywiki/filesystem"
   ],
   "themes": [
      "tiddlywiki/vanilla"
   ]
}
````

2. Added two parts of your plugin : 
    * $__plugins_felixhayashi_tiddlymap.tid
    * $__plugins_felixhayashi_vis.tid 

3. Launch the server
 `tiddlywiki ./Test --server` and open the browser at 127.0.0.1:8080 so the tiddlywiki is running

4. Here comes the silly part that might make your plugin innocent
__in the text editor__ (server still running) create a new tiddler with a title, a body and any other field but no tmap.id.

5. kill server by closing the terminal, wait until your browser notice there is no more server `Error retrieving skinny tiddler list: XMLHttpRequest error code: 0` then relaunch (cf. point 3)

6. refresh the tiddlywiki in your browser
Here's the terminal mentions : 
````
Serving on 127.0.0.1:8080
(press ctrl-C to exit)
syncer-server: Dispatching 'save' task: $:/StoryList
FileSystem: Saved file (...)\Test\tiddlers\$__StoryList.tid
syncer-server: Dispatching 'save' task: another tiddler
FileSystem: Saved file (...)\Test\tiddlers\another.tid
````

And now __sometimes__ (but why sometimes, for now and after many attempts i didn't get it) the tiddler has no more body. All fields were preserved except `modified` of course but body disappeared... When tiddler contains a manually added tmap.id field, the save for the tiddler is not launched at startup and the body never destroyed.

**Comment by felixhayashi** [comment]
Hmm, reading this strongly reminds me of the bug I just fixed today (see #222). In this case, a wiki running via a nodejs server in two browser windows would result in a loss of a tiddler body as soon as a tiddler is saved in one window while the other window tries to sync the tiddler.

However, I just noticed, I made a mistake in the bugfix that came with `v0.12.1`… but `v0.12.2` should also fix your issue because past versions of TiddlyMap assigned an empty text to every changed tiddler that came without a text field (which is the case for every synced tiddler since these are "skinny", i.e. without body).

I'll close this issue now, please reopen if its still not working for you.

-Felix
